### PR TITLE
Provenance Spec improvements v0.2

### PR DESCRIPTION
**Pull request by aaitor**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: docs/architecture/specs/provenance/README.md [hunk 1]
```diff
@@ -56,8 +67,22 @@ The main objectives of this SPEC are:
...
+## Concepts
+
+### Specifications
+
+Nevermined Data Provenance solution is designed based on the [W3C Provenance specifications](https://www.w3.org/TR/2013/NOTE-prov-overview-20130430/).
+In addition to this, Nevermined uses [Decentralized Identifiers](https://keyko-io.github.io/nevermined-docs/#/architecture/specs/did/README)
+following the [W3C DID specification](https://w3c.github.io/did-core/#overall-architecture).
 
-## Decentralized Data Provenance
+The design of Nevermined using both specifications is critical because enable the
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:78]: `because it enables`?

File: docs/architecture/specs/provenance/README.md [hunk 2]
```diff
@@ -107,7 +130,9 @@ first version of the document in another language, then this translation is an a
 
 #### Provenance Relations
 
-##### Generation: Register a new entity generated by an agent
+Relations between agents and entities are open but there some common relations that can be used as reference:
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:133]:
```suggestion
Relations between agents and entities are open but there are some common relations that can be used as reference:
```

File: docs/architecture/specs/provenance/README.md [hunk 3]
```diff
@@ -193,6 +218,79 @@ underpinning the derivation, it can identify how the preceding entity `did:nv:e1
...
+ actedOnBehalfOf(did:nv:a1, did:nv:a2, [ prov:type="ground transportation" ])
+```
+
+
+#### Usage (used)
+
+`Usage` is the beginning of utilizing an entity by an activity.
+Before usage, the activity had not begun to utilize this entity and could not
+have been affected by the entity. (Note: This definition is formulated for a
+given usage; it is permitted for an activity to have used a same entity multiple
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:272]:
```suggestion
given usage; it is permitted for an activity to use the same entity multiple
```

File: docs/architecture/specs/provenance/README.md [hunk 4]
```diff
@@ -193,6 +218,79 @@ underpinning the derivation, it can identify how the preceding entity `did:nv:e1
...
+
+Usage, written `used(id; a, e, t, attrs)` in PROV-N, has:
+
+* `id`: an optional identifier for a usage;
+* `activity`: an identifier (a) for the activity that used an entity;
+* `entity`: an optional identifier (e) for the entity being used;
+* `time`: an optional "usage time" (t), the time at which the entity started to be used;
+* `attributes`: an optional set (attrs) of attribute-value pairs representing additional information about this usage.
+
+In the following example we can see how is associated the activity `did:nv:ac1`
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:285]:
```suggestion
In the following example we can see how to associate the activity `did:nv:ac1`
```

File: docs/architecture/specs/provenance/README.md [hunk 5]
```diff
@@ -193,6 +218,79 @@ underpinning the derivation, it can identify how the preceding entity `did:nv:e1
...
+with the usage of the entity `did:nv:e1` in sone specific moment.
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:286]:
```suggestion
with the usage of the entity `did:nv:e1` in some specific moment.
```

File: docs/architecture/specs/provenance/README.md [hunk 6]
```diff
@@ -216,228 +314,230 @@ activity(a1, 2011-11-16T16:05:00, 2011-11-16T16:06:00,
...
-    },
-    "activity": { // Map of activities by IDs
-    },
-    "agent": { // Map of agents by IDs
-    },
-    <relationName>: { // A map of relations of type relationName by their IDs
-    }
-}
-```
+For describing the architecture we are gonna use as a reference a supply chain chain use case. In this scenario a company manufacturing some goods needs to deliver them to the final customer. To do that the goods need to be transported by different shipping companies until the goods are handed over to the final recipient.
```
Comment (r-marques) [inline docs/architecture/specs/provenance/README.md:343]:
```suggestion
For describing the architecture we are gonna use as a reference a supply chain use case. In this scenario a company manufacturing some goods needs to deliver them to the final customer. To do that the goods need to be transported by different shipping companies until the goods are handed over to the final recipient.  
```

**Review by r-marques — APPROVED** [review]
left some comments but lgtm

**Comment by aaitor** [comment]
thanks for your comments @r-marques 
I applied the changes suggested

**Pull request merged**